### ISSUE TITLE
4.0.0-preview-0651 - Expand single endpoint in UI expands all in list

### ISSUE THREAD

**Issue by carl-berg** [domaindrivendev/Swashbuckle.AspNetCore#904]
Something i noticed when i switched from last stable version to _4.0.0-preview-0651_ was that when clicking on an endpoint, all endpoints in the list expanded. I would assume the expected behavior is that only the clicked one would expand.

**Comment by domaindrivendev** [comment]
TLDR; If you're only leveraging Swashbuckle for the UI functionality, and there's no other consumers of the generated Swagger JSON, then you can resolve this issue by adding the following to your `SwaggerGen` config:

```csharp
services.AddSwaggerGen(c =>
{
    …

    c.CustomOperationIds(apiDesc => null);
}
```

If you do have other consumers, particularly client generators, or you just want a further explanation, read on ...

This is an unforseen side affect of a breaking change that I intentionally introduced with version 4.0.0 ([release notes here](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/releases/tag/v4.0.0)). Unlike previous versions, Swashbuckle no longer auto-generates the `operationId` field for actions. This field is typically used in tools that generate client libraries (AutoRest, NSwag etc.). It's described in the OpenAPI specification as follows:

>The id MUST be unique among all operations described in the API. Tools and libraries MAY use the operationId to uniquely identify an operation, therefore, it is RECOMMENDED to follow common programming naming conventions.

Auto-generating an ID that matches these requirements, while also providing a name that would be meaningful in client libraries has proved to be extremely difficult (see #333, #876), and so I made the explicit decision to remove this capability and instead put the onus on API authors to come up with unique and meaningful `operationIds`.

However, I decided to use the method name by default, and the route name if provided (e.g. `[HttpPost("/api/products", Name = "CreateProduct")]`). See [the readme topic](https://github.com/domaindrivendev/Swashbuckle.AspNetCore#assign-explicit-operationids) for more details. So, if you're not providing a route name, and your method names aren't unique across all actions in your API (a fairly likely scenario), then the resulting Swagger will have conflicting operationIds, and this will cause the UI behavior you're observing.

In hindsight, I think Swashbuckle should only set the `operationId` if a value has been explicitly provided via the route name. It's not a required field and the Swagger UI functions just fine without it (hence the quick fix described above). Of course this MAY break some client generators, particularly if they depend on the presence of an `operationId`. But, the Swagger UI is a very common Swashbuckle use-case and so I'd like to optimize for that. If folks are using client generators, then they'll need to provide an explicit `Name` with their route declarations.

**Comment by domaindrivendev** [comment]
See 6d4beaf859f7a3407de0b06bee14c4b4c9c508a7
